src/EditorFeatures/CSharpTest/TypeInferrer/TypeInferrerTests.cs

```diff
@@ -3047,7 +3047,7 @@ static void Main(string[] args)
     static T Identity<T>(T value) { return value; }
 }";
 
-            await TestAsync(text, "global::System.String?");
+            await TestAsync(text, "global::System.String?", testPosition: false);
```

Comment:
@CyrusNajmabadi What's the logic behind when tests should be passing `testPosition: false` or `testNode: false`? I see a bunch of tests doing it but no logic as to _why_.

Reply:
And maybe to clarify: yes, I see one is passing a node's span and one is passing just a position, but I don't have an intuition as far as why some scenarios work one way versus the other.

Reply:
Chatting with @CyrusNajmabadi: when inferring you should think of the two modes of being "what would the type of this be if this identifier did exist" and "what would the type of the thing be if I started typing here". Usually these are the same but an example was:

    bool b = $$i > 0;

In this case, positionally we would say "bool" (because we're not really worrying about what is already there to the right of the cursor) but the node case would say "int" because `i` would be inferred to be an int.

In this case, there's no reason for these to be different so I've now fixed the bug.